modules/rollup-plugin-workbox-inject/index.js

```diff
@@ -31,39 +33,74 @@ const chalk = require('chalk');
 function workboxInject(config) {
   return {
     name: 'workbox-inject', // this name will show up in warnings and errors
-    async generateBundle(opts, bundle) {
-      const options = validate(config, injectManifestSchema);
-      // Make sure we leave swSrc and swDest out of the precache manifest.
-      for (const file of [options.swSrc, options.swDest]) {
-        options.globIgnores.push(
-          rebasePath({
-            file,
+    async generateBundle(outputOptions, bundle) {
+      // injectionPoint isn't supported by getManifestSchema, but we need it.
+      const injectionPoint = config.injectionPoint || defaults.injectionPoint;
+      delete config.injectionPoint;
+
+      const options = validate(config, getManifestSchema);
+
+      // globDirectory might not be set, as developers can just provide
+      // additionalManifestEntries and forgo globbing.
+      if (options.globDirectory) {
+        const outputDir = outputOptions.dir ||
+          upath.dirname(upath.resolve(outputOptions.file));
+
+        // Make sure we leave the source service worker and any output of this
+        // specific Rollup compilation out of the precache manifest.
+        for (const chunk of Object.values(bundle)) {
+          const fileRebasedToGlobDirectory = rebasePath({
+            file: upath.resolve(outputDir, chunk.fileName),
             baseDirectory: options.globDirectory,
-          }),
-        );
+          });
+          // Sourcemaps might be disabled, but adding something extra to
+          // globIgnores doesn't hurt.
+          const mapFile = `${fileRebasedToGlobDirectory}.map`;
+
+          options.globIgnores.push(
+            fileRebasedToGlobDirectory,
+            mapFile,
+          );
+
+          // If set, this should be the path to the original source module.
+          if (chunk.facadeModuleId) {
+            const moduleRebasedToGlobDirectory = rebasePath({
+              file: chunk.facadeModuleId,
+              baseDirectory: options.globDirectory,
+            });
+            options.globIgnores.push(moduleRebasedToGlobDirectory);
+          }
+        }
       }
-      const globalRegexp = new RegExp(escapeRegexp(options.injectionPoint), 'g');
 
       const { manifestEntries, count, warnings, size } = await getFileManifestEntries(options);
 
       const manifestString = stringify(manifestEntries);
 
-      const bundles = Object.keys(bundle);
+      for (const chunk of Object.values(bundle)) {
+        // This might actually be an asset (which won't have a code property)
+        // and not a chunk. If so, continue.
+        if (!chunk.code) {
```

Comment:
Bug fix: in some scenarios, the `bundle` might contain an `asset` and not `chunk`, so you can't assume that there's always going to be a `code` property.